code/obj/mining.dm

```diff
@@ -692,6 +701,8 @@
 		.["time"] = TIME
 
 	ui_act(action, params)
+		if (src.check_should_die())
+			return
```

Comment:
```suggestion
```
Things should explode when they are exploded, not when the UI code executes. If this check is necessary then look into why and fix it.